ui/helpers/utils/token-util.js

```diff
@@ -212,3 +215,55 @@ export function getTokenFiatAmount(
   }
   return result;
 }
+
+export async function getAssetDetails(
+  tokenAddress,
+  currentUserAddress,
+  transactionData,
+  existingCollectibles,
+) {
+  const tokenData = parseTransactionData(transactionData);
+  if (!tokenData) {
+    throw new Error('Unable to detect valid token data');
+  }
+
+  const tokenId = getTokenValueParam(tokenData);
+  let tokenDetails;
+  try {
+    tokenDetails = await getTokenStandardAndDetails(
+      tokenAddress,
+      currentUserAddress,
+      tokenId,
+    );
+  } catch (error) {
+    log.warn(error);
+    return {};
+  }
+  if (tokenDetails?.standard === ERC721 || tokenDetails?.standard === ERC1155) {
+    const existingCollectible = existingCollectibles.find(({ address }) =>
+      util.isEqualCaseInsensitive(tokenAddress, address),
+    );
+    if (existingCollectible) {
+      return {
+        address: existingCollectible?.address,
+        description: existingCollectible?.description,
+        favorite: existingCollectible?.favorite,
+        image: existingCollectible?.image,
+        isCurrentlyOwned: existingCollectible?.isCurrentlyOwned,
+        name: existingCollectible?.name,
+        tokenId: existingCollectible?.tokenId,
+        standard: tokenDetails?.standard,
+      };
+    }
+    return tokenDetails;
+  } else if (tokenDetails?.standard === ERC20) {
+    return {
+      symbol: tokenDetails?.symbol,
+      decimals: tokenDetails?.decimals,
+      balance: tokenDetails?.balance,
+      standard: tokenDetails?.standard,
+    };
```

Comment:
I think that `tokenDetails?.standard === ERC20` validates that `tokenDetails` exists, so I'm not sure you need `?.` on the properties either.